--- a/configure.in
+++ b/configure.in
@@ -1840,13 +1840,6 @@
 fi
 final_message="$final_message\n  SWIG Available:\tYes"
 
-echo "Checking for a jre installation"
-AC_PATH_PROG(JAVA_EXE, java, "none")
-if test "$JAVA_EXE" = "none"; then
-  AC_MSG_ERROR($missing_program)
-fi
-final_message="$final_message\n  JRE Available:\tYes"
-
 echo "Checking for doxygen installation"
 AC_PATH_PROG(DOXYGEN_EXE, doxygen, "none")
 if test "$DOXYGEN_EXE" = "none"; then
