--- a/pvr-addons/configure.ac
+++ b/pvr-addons/configure.ac
@@ -124,12 +124,6 @@
 esac
 
 ### External libraries checks
-AC_CHECK_LIB([GL],[main],have_gl=yes, have_gl=no)  
-if test "$have_gl" = "yes"; then
-  AC_DEFINE([HAVE_GL],[1],["Define to 1 if we have gl"])
-fi
-echo "GL: $have_gl"
-
 AC_CHECK_LIB([GLESv2],[main],[AC_CHECK_HEADER([GLES2/gl2.h], have_gles2=yes, have_gles2=no, )], have_gles2=no)
 if test "$have_gles2" = "yes"; then
   AC_DEFINE([HAVE_GLES2],[1],["Define to 1 if we have gles2"])
@@ -137,8 +131,18 @@
 else
   AM_CONDITIONAL(HAVE_GLES2, false)
 fi
-echo "GLES2: $have_gles2"
 
+AC_CHECK_LIB([GL],[main],have_gl=yes, have_gl=no)  
+if test "$have_gl" = "yes"; then
+  AC_DEFINE([HAVE_GL],[1],["Define to 1 if we have gl"])
+fi
+
+if test "$have_gles2" = "yes"; then
+  have_gl=no
+fi
+
+echo "GL: $have_gl"
+echo "GLES2: $have_gles2"
 ### End external Libraries 
 
 ### Build addons with dependencies
