--- a/addons/webinterface.default/js/MediaLibrary.js
+++ b/addons/webinterface.default/js/MediaLibrary.js
@@ -228,12 +228,12 @@
             {'playerid': player, 'to': 'previous'}
           );
         case 'forward':
-          return this.rpcCall('Player.SetSpeed',
-            {'playerid': player, 'speed': 'increment'}
+          return this.rpcCall('Player.Seek',
+            {'playerid': player, 'speed': 'smallforward'}
           );
         case 'rewind':
-          return this.rpcCall('Player.SetSpeed',
-            {'playerid': player, 'speed': 'decrement'}
+          return this.rpcCall('Player.Seek',
+            {'playerid': player, 'speed': 'smallbackward'}
           );
       }
     }
