--- a/xbmc/interfaces/legacy/ModuleXbmcgui.h	2014-06-05 14:48:24.000000000 +0300
+++ b/xbmc/interfaces/legacy/ModuleXbmcgui.h	2014-06-06 23:36:53.042113692 +0300
@@ -60,6 +60,42 @@
      */
     long getCurrentWindowDialogId();
 
+    /**
+     * getMouseRawPosition() -- Returns the mouse raw coordinats as an integer.
+     * 
+     * example:
+     *   - pos = xbmcgui.getMouseRawPosition()
+     *     x = pos / 10000
+     *     y = pos % 10000
+     */ 
+    long getMouseRawPosition();
+
+    /**
+     * getMousePosition() -- Returns the mouse calibrated coordinats as an integer.
+     * 
+     * example:
+     *   - pos = xbmcgui.getMousePosition()
+     *     x = pos / 10000
+     *     y = pos % 10000
+     */ 
+    long getMousePosition();
+
+    /**
+     * getMouseRawAction() -- Returns the raw mouse state.
+     *
+     * example:
+     *   - act = xbmcgui.getMouseRawAction()
+     */
+    long getMouseRawAction();
+
+    /**
+     * getMouseAction() -- Returns the mouse state.
+     *
+     * example:
+     *   - act = xbmcgui.getMouseAction()
+     */
+    long getMouseAction();
+
     SWIG_CONSTANT2(int,ICON_OVERLAY_NONE, CGUIListItem::ICON_OVERLAY_NONE);
     SWIG_CONSTANT2(int,ICON_OVERLAY_RAR, CGUIListItem::ICON_OVERLAY_RAR);
     SWIG_CONSTANT2(int,ICON_OVERLAY_ZIP, CGUIListItem::ICON_OVERLAY_ZIP);
