--- a/xbmc/interfaces/legacy/ModuleXbmcgui.cpp	2014-04-27 07:55:16.000000000 +0300
+++ b/xbmc/interfaces/legacy/ModuleXbmcgui.cpp	2014-08-08 21:29:29.401030443 +0300
@@ -22,6 +22,7 @@
 #include "LanguageHook.h"
 #include "guilib/GraphicContext.h"
 #include "guilib/GUIWindowManager.h"
+#include "input/MouseStat.h"
 #include "utils/log.h"
 
 #define NOTIFICATION_INFO     "info"
@@ -55,7 +56,27 @@
       CSingleLock gl(g_graphicsContext);
       return g_windowManager.GetTopMostModalDialogID();
     }
+
+    long getMouseRawPosition()
+    {
+      return g_Mouse.GetRawX() * 10000 + g_Mouse.GetRawY();
+    }
+    
+    long getMousePosition()
+    {
+      return g_Mouse.GetX() * 10000 + g_Mouse.GetY();
+    }
     
+    long getMouseRawAction()
+    {
+    	return g_Mouse.GetRawAction();
+    }
+
+    long getMouseAction()
+    {
+    	return g_Mouse.GetAction();
+    }
+
     const char* getNOTIFICATION_INFO()    { return NOTIFICATION_INFO; }
     const char* getNOTIFICATION_WARNING() { return NOTIFICATION_WARNING; }
     const char* getNOTIFICATION_ERROR()   { return NOTIFICATION_ERROR; }
