--- a/xbmc/windowing/egl/EGLQuirks.h
+++ b/xbmc/windowing/egl/EGLQuirks.h
@@ -39,3 +39,8 @@
            size of -1, -1. Work around it for now
 */
 #define EGL_QUIRK_DONT_TRUST_SURFACE_SIZE (1 << 2)
+
+/*! \brief Enable this to avoid releasing the EGL context
+           This avoids a problem with texture creation failing when using "virtual suspend"
+*/
+#define EGL_QUIRK_KEEP_CONTEXT (1 << 3)
--- a/xbmc/windowing/egl/WinSystemEGL.cpp
+++ b/xbmc/windowing/egl/WinSystemEGL.cpp
@@ -289,12 +296,13 @@ bool CWinSystemEGL::DestroyWindow()
   if (!m_egl)
     return false;
 
-  m_egl->ReleaseContext(m_display);
-  if (m_surface != EGL_NO_SURFACE)
-    m_egl->DestroySurface(m_surface, m_display);
-
   int quirks;
   m_egl->GetQuirks(&quirks);
+
+  if (!(quirks & EGL_QUIRK_KEEP_CONTEXT))
+    m_egl->ReleaseContext(m_display);
+  if (m_surface != EGL_NO_SURFACE)
+    m_egl->DestroySurface(m_surface, m_display);
   if (quirks & EGL_QUIRK_DESTROY_NATIVE_WINDOW_WITH_SURFACE)
     m_egl->DestroyNativeWindow();
 
