--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -237,18 +237,17 @@
   std::string valstr;
   get_sysfs_str("/sys/class/graphics/fb0/modes", valstr);
 
-  std::vector<CStdString> probe_str;
-  StringUtils::SplitString(valstr, "\n", probe_str);
+  std::vector<std::string> probe_str = StringUtils::Split(valstr, "\n");
   std::sort(probe_str.begin(), probe_str.end());
 
   resolutions.clear();
   RESOLUTION_INFO res;
-  for (size_t i = 0; i < probe_str.size(); i++)
+  for (std::vector<std::string>::const_iterator i = probe_str.begin(); i != probe_str.end(); ++i)
   {
-    if(!StringUtils::StartsWith(probe_str[i], "S:") && !StringUtils::StartsWith(probe_str[i], "U:"))
+    if(!StringUtils::StartsWith(*i, "S:") && !StringUtils::StartsWith(*i, "U:"))
       continue;
 
-    if(ModeToResolution(probe_str[i], &res))
+    if(ModeToResolution(*i, &res))
       if(!FindMatchingResolution(act, res, resolutions))
          resolutions.push_back(res);
   }
