diff --git a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index 149591c..3cf713c 100644
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -19,6 +19,7 @@
  */
 
 #include <EGL/egl.h>
+#include "system.h"
 
 #include "EGLNativeTypeIMX.h"
 #include <math.h>
@@ -64,6 +65,7 @@ void CEGLNativeTypeIMX::Initialize()
 
   CSettings::Get().SetInt("audiooutput.streamsilence", 0);
 
+  system("/usr/bin/splash --force -b");
   // Check if we can change the framebuffer resolution
   if ((fd = open("/sys/class/graphics/fb0/mode", O_RDWR)) >= 0)
   {
--- a/xbmc/xbmc.cpp
+++ b/xbmc/xbmc.cpp
@@ -74,10 +74,6 @@
     return status;
   }
     
-  if (!showXBMCSplash) {
-    system("/usr/bin/splash --force --black");
-  }
-
   try
   {
     status = g_application.Run();
