diff --git a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
index 8492459..7e66328 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
@@ -1007,6 +1020,7 @@ void CLinuxRendererGLES::Render(DWORD flags, int index)
   else if (m_renderMethod & RENDER_IMXMAP)
   {
     RenderIMXMAPTexture(index, m_currentField);
+    VerifyGLState();
   }
   else
   {
@@ -2802,7 +2816,7 @@ bool CLinuxRendererGLES::CreateIMXMAPTexture(int index)
   YUVFIELDS &fields = m_buffers[index].fields;
   YUVPLANE  &plane  = fields[0][0];
 
-  DeleteEGLIMGTexture(index);
+  DeleteIMXMAPTexture(index);
 
   memset(&im    , 0, sizeof(im));
   memset(&fields, 0, sizeof(fields));
