commit ff50213c4b9e09de47fcbf3448d6c373326781ff
Author: Matus Kral <matuskral@me.com>
Date:   Tue Jul 29 14:59:00 2014 +0200

    allows both ALSA + PULSEAUDIO sinks

diff --git a/xbmc/cores/AudioEngine/AESinkFactory.cpp b/xbmc/cores/AudioEngine/AESinkFactory.cpp
index e493123..f49853f 100644
--- a/xbmc/cores/AudioEngine/AESinkFactory.cpp
+++ b/xbmc/cores/AudioEngine/AESinkFactory.cpp
@@ -246,7 +246,7 @@ void CAESinkFactory::EnumerateEx(AESinkInfoList &list, bool force)
   if(!info.m_deviceInfoList.empty())
   {
     list.push_back(info);
-    return;
+//    return;
   }
   #endif
 
@@ -257,7 +257,7 @@ void CAESinkFactory::EnumerateEx(AESinkInfoList &list, bool force)
   if(!info.m_deviceInfoList.empty())
   {
     list.push_back(info);
-    return;
+//    return;
   }
   #endif
 
diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp
index 4923614..53e62bd 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp
@@ -431,6 +431,13 @@ static void SinkInfoRequestCallback(pa_context *c, const pa_sink_info *i, int eo
       device.m_deviceType = AE_DEVTYPE_IEC958;
     else
       device.m_deviceType = AE_DEVTYPE_PCM;
+
+    if (device.m_deviceType == AE_DEVTYPE_PCM && device.m_displayName.find("hdmi") != std::string::npos)
+      device.m_deviceType = AE_DEVTYPE_HDMI;
+    else if (device.m_deviceType == AE_DEVTYPE_PCM &&
+            (device.m_displayName.find("iec958") != std::string::npos || device.m_displayName.find("spdif") != std::string::npos))
+      device.m_deviceType = AE_DEVTYPE_IEC958;
+
     if(valid)
     {
       CLog::Log(LOGDEBUG, "PulseAudio: Found %s with devicestring %s", device.m_displayName.c_str(), device.m_deviceName.c_str());
