--- a/configure.in
+++ b/configure.in
@@ -1618,9 +1618,6 @@ final_message="$final_message\n  SWIG Av

 echo "Checking for a jre installation"
 AC_PATH_PROG(JAVA_EXE, java, "none")
-if test "$JAVA_EXE" = "none"; then
-  AC_MSG_ERROR($missing_program)
-fi
 final_message="$final_message\n  JRE Available:\tYes"

 echo "Checking for doxygen installation"
--- a/codegenerator.mk
+++ b/codegenerator.mk
@@ -53,11 +53,6 @@
    @echo "Warning: No doxygen installed. The Api will not have any docstrings."
    mkdir -p $(GENDIR)/doxygenxml
 
-$(JAVA):
-	@echo Java not found, it will be used if found after configure.
-	@echo This is not necessarily an error.
-	@false
-
 $(SWIG):
    @echo Swig not found, it will be used if found after configure.
    @echo This is not necessarily an error.
