--- a/xbmc/xbmc.cpp	
+++ b/xbmc/xbmc.cpp
@@ -27,6 +27,8 @@
 #include "linux/RBP.h"
 #endif
 
+#include "utils/log.h"
+
 extern "C" int XBMC_Run(bool renderGUI)
 {
   int status = -1;
@@ -64,6 +76,9 @@ extern "C" int XBMC_Run(bool renderGUI)
     return status;
   }
 
+  CLog::Log(LOGNOTICE, "XBian: notifying Upstart that i'm well");
+  system("sudo /sbin/start -n -q xbmc-loaded");
+
   try
   {
     status = g_application.Run();
