--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -5236,8 +5236,8 @@
 {
   // This 'if' clause is completely unnecessary, but by not touching the 'return'
   // below this patch becomes more easily maintainable.
-  if (GetCecStandby())
-    return true;
+//  if (GetCecStandby())
+//    return true;
 
   return IsInScreenSaver() && m_screenSaver &&
     (m_screenSaver->ID() == "screensaver.xbmc.builtin.black" ||
