--- a/configure.in
+++ b/configure.in
@@ -1731,6 +1731,13 @@
     if test "$use_optimizations" != "yes"; then
       FFMPEG_OPTS="${FFMPEG_OPTS} --disable-optimizations"
     fi
+    FFMPEG_OPTS="${FFMPEG_OPTS} --"
+    if test "$use_neon" != "yes"; then
+      FFMPEG_OPTS="${FFMPEG_OPTS} --disable-neon"
+    fi
+    if test "$use_neon" != "yes"; then
+      FFMPEG_OPTS="${FFMPEG_OPTS} --enable-hardcoded-tables"
+    fi
 
     if test "$with_ffmpeg" = "auto" || test "$with_ffmpeg" = "yes"; then
       SAVE_INCLUDES="$INCLUDES"
