--- a/tools/depends/native/libsquish-native/Makefile
+++ b/tools/depends/native/libsquish-native/Makefile
@@ -27,7 +28,8 @@
 	@echo "ERROR: please set PREFIX to the xbmc install path e.g. make PREFIX=/usr/local"
 	@exit 1
 endif
-	-rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
+	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
 	cd $(PLATFORM); cp -a $(SOURCE)/* .
 	cd $(PLATFORM); sed "s|@PREFIX@|$(PREFIX)|" config.in > config
+	mkdir -p $(PREFIX)
 	cd $(PLATFORM); mkdir -p $(PREFIX)/include; mkdir -p $(PREFIX)/lib/pkgconfig
 
 $(DYLIB): $(PLATFORM)
