diff --git b/lib/ffmpeg/libavcodec/arm/dca.h a/lib/ffmpeg/libavcodec/arm/dca.h
index 431b62e..729127a 100644
--- b/lib/ffmpeg/libavcodec/arm/dca.h
+++ a/lib/ffmpeg/libavcodec/arm/dca.h
@@ -30,9 +30,9 @@
 
 #define decode_blockcodes decode_blockcodes
 static inline int decode_blockcodes(int code1, int code2, int levels,
-                                    int *values)
+                                    int32_t *values)
 {
-    int v0, v1, v2, v3, v4, v5;
+    int32_t v0, v1, v2, v3, v4, v5;
 
     __asm__ ("smmul   %0,  %6,  %10           \n"
              "smmul   %3,  %7,  %10           \n"
