commit c8c1572cf50247336ed3e063c66c54e49493c49b
Author: Matus Kral <matuskral@me.com>
Date:   Tue Aug 5 01:25:39 2014 +0200

  allow stereoscopic modes handling

diff --git a/xbmc/cores/VideoRenderers/BaseRenderer.cpp b/xbmc/cores/VideoRenderers/BaseRenderer.cpp
index 970b822..52dd8e7 100644
--- a/xbmc/cores/VideoRenderers/BaseRenderer.cpp
+++ b/xbmc/cores/VideoRenderers/BaseRenderer.cpp
@@ -581,7 +604,7 @@ void CBaseRenderer::ManageDisplay()
   {
     case CONF_FLAGS_STEREO_MODE_TAB:
       // Those are flipped in y
-      if (m_format == RENDER_FMT_CVBREF || m_format == RENDER_FMT_EGLIMG || m_format == RENDER_FMT_MEDIACODEC)
+      if (m_format == RENDER_FMT_IMXMAP || m_format == RENDER_FMT_CVBREF || m_format == RENDER_FMT_EGLIMG || m_format == RENDER_FMT_MEDIACODEC)
       {
         if (stereo_view == RENDER_STEREO_VIEW_LEFT)
           m_sourceRect.y1 += m_sourceRect.y2*0.5f;
